### PR TITLE
BUG: cut with non-nano

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

sits on top of #56099

### CODE REVIEW

File: pandas/tests/reshape/test_cut.py [hunk 1]
```diff
@@ -576,17 +564,32 @@ def test_datetime_nan_mask():
...
     tm.assert_series_equal(result, expected)
 
-    expected_bins = DatetimeIndex(
-        ["2017-12-31 23:57:07.200000", "2018-01-02 00:00:00", "2018-01-03 00:00:00"]
-    )
+    if unit == "s":
+        # TODO: constructing DatetimeIndex with dtype="M8[s]" without truncating
+        #  the first entry here raises in array_to_datetime. Should truncate
+        #  instead of raising?
+        # TODO: should we truncate to 7 seconds or 8?
```
Comment (mroeschke) [inline pandas/tests/reshape/test_cut.py:579]: If these are inclusive, left bin edge and in nanos the result was 7.2, I think I would expect 7 instead of 8

Comment (jbrockmendel) [reply pandas/tests/reshape/test_cut.py]: looking at this closer, i think it doesn't _really_ matter bc `bins[0]` is calculated roughly on `min(data) - .001 * (max(data) - min(data))`.  that `.001 * [...]` adjustment looks intended to make sure that the bottom bin is a strict lower bound.  Exactly how strict doesn't seem to matter.  In this test case, max(data) - min(data) is a 2-day Timedelta, and when that Timedelta has second unit, multiplying it by .001 truncated to `Timedelta(minutes=2, seconds=52)`, truncating the .8 seconds.

Comment (mroeschke) [reply pandas/tests/reshape/test_cut.py]: Ah I forgot about the 0.001 offset applied to the edges (though it seem code-smelly). The current truncation result should be OK then

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**